Automattic/wp-calypso
# Initial work to add `Request Transfer` buttons to the UI

#### Changes proposed in this Pull Request

* Add a `Transfer Domain to WordPress.com` action for domains that are mapped in from elsewhere.

#### Testing instructions

* Verify that when you view the details for a mapped domain, a "Transfer Domain to WordPress.com" item appears in the nav options at the bottom of the screen.
  * When you click on that item, you should shift to the Transfer Domain UI with the domain entered in the search/input box.
  * When you click the "Back" link in the header, ensure that you go back to the domain management screen for the mapped domain.
* Verify that the link _does not_ appear for a mapped subdomain.
* Verify that when you try to add a domain, and then work through the "Use a domain I own" -> "Transfer to WordPress.com" flow, and then click "Back" in the header, you should be returned to the domain search screen.

### Open Issues/Questions

* Please take a **very close look** at the way I've handled the "Back" button logic, which involves a new URL parameter _and_ a call to `page.back()`. The original logic was very stateful, and I've added a less stateful approach, but it doesn't change the original logic (which dates back to the original implementation for domain transfers).

**NOTE:** This depends on back end change D33690-code.
